lib/mix/test/mix/dep_test.exs

```diff
@@ -266,6 +266,31 @@ defmodule Mix.DepTest do
     end)
   end
 
+  test "deps with env_vars set" do
+    dep_path = MixTest.Case.tmp_path("rebar_dep")
+
+    with_deps(
+      [
+        {
```

Comment:
Please look at the other tests. You should define the deps as a variable before calling `with_deps`.